astropy/io/fits/tests/test_table.py

```diff
@@ -1702,12 +1702,14 @@ def test_multi_dimensional_columns(self):
              ([2, 3, 4, 5, 6, 7], 'row3' * 2)], formats='6i4,a8')
 
         thdu = fits.BinTableHDU.from_columns(data)
-        # Modify the TDIM fields to my own specification
-        thdu.header['TDIM1'] = '(2,3)'
-        thdu.header['TDIM2'] = '(4,2)'
 
         thdu.writeto(self.temp('newtable.fits'))
 
+        with fits.open(self.temp('newtable.fits'), mode='update') as hdul:
+            # Modify the TDIM fields to my own specification
+            hdul[1].header['TDIM1'] = '(2,3)'
+            hdul[1].header['TDIM2'] = '(4,2)'
+
```

Comment:
Why is this change needed ?

Reply:
At least when running the tests locally, it seemed that setting these headers *before* writing the file was overwritten by the actual dimensions of the data.

So I had to change to modify them after the file was initially written.

Reply:
This did not trigger before, as TDIM was only added `if len(shape) > 1`, which is no longer the case.